### ISSUE TITLE
File Chooser(From Locus) for map file selection not starting

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#8525]
**Describe the bug:**
File chooser not starting for map file selection

**To Reproduce:**
1. Go to Menu - Settings - Map
2. Click on "Directory  with offline maps"
3. After scanning finished, click on "Choose Directory"

**Actual behavior/state after performing these steps:**
c:geo jumps back to settings menu. File chooser not shown

**Expected behavior/state after performing these steps:**
File chooser should be started

**Version of c:geo used:**
2020.06.22-NB

**Is the problem reproducible:**
Yes 

**Additional context:**
Was also reported on support mail, still waiting for more details (c:geo version, logfile) from user, but probably same problem

After several tries my system reports "Locus Map stopped repeatedly". So I assume its a problem in connection with the file chooser which we use coming from Locus.

**Comment by Lineflyer** [comment]
It is also happening on release and seems to be related to having Locus installed (and we use their file browser).

Stack trace:
```
21:24:49.160 Error AndroidRuntime 15140  java.lang.RuntimeException: Unable to start activity ComponentInfo{menion.android.locus.pro/com.asamm.locus.gui.activities.fileBrowser.FileBrowser}: kotlin.KotlinNullPointerException
21:24:49.160 Error AndroidRuntime 15140  	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2702)
21:24:49.160 Error AndroidRuntime 15140  	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2767)
21:24:49.160 Error AndroidRuntime 15140  	at android.app.ActivityThread.access$900(ActivityThread.java:177)
21:24:49.160 Error AndroidRuntime 15140  	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1449)
21:24:49.160 Error AndroidRuntime 15140  	at android.os.Handler.dispatchMessage(Handler.java:102)
21:24:49.160 Error AndroidRuntime 15140  	at android.os.Looper.loop(Looper.java:145)
21:24:49.160 Error AndroidRuntime 15140  	at android.app.ActivityThread.main(ActivityThread.java:5951)
21:24:49.160 Error AndroidRuntime 15140  	at java.lang.reflect.Method.invoke(Native Method)
21:24:49.160 Error AndroidRuntime 15140  	at java.lang.reflect.Method.invoke(Method.java:372)
21:24:49.160 Error AndroidRuntime 15140  	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
21:24:49.160 Error AndroidRuntime 15140  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
21:24:49.160 Error AndroidRuntime 15140  Caused by: kotlin.KotlinNullPointerException
21:24:49.160 Error AndroidRuntime 15140  	at o.вΙ.<init>(:90)
21:24:49.160 Error AndroidRuntime 15140  	at com.asamm.locus.gui.activities.fileBrowser.FileBrowser.onCreate(:81)
21:24:49.160 Error AndroidRuntime 15140  	at android.app.Activity.performCreate(Activity.java:6289)
21:24:49.160 Error AndroidRuntime 15140  	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
21:24:49.160 Error AndroidRuntime 15140  	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2655)
21:24:49.160 Error AndroidRuntime 15140  	... 10 more
```

Will try to file a report to Locus.

**Comment by Lineflyer** [comment]
Reported to help.locusmap.eu

**Comment by Swoddler** [comment]
I have this problem too. Cannot change my offline maps because I can't get to the directory stage. Also getting the 'Locus' message (I do have Locus installed and do need it for ordnance survey maps). Any fixes or workarounds?

- Most up to date vesion of cgeo on a Samsung A70

**Comment by moving-bits** [comment]
@Swoddler 

If you're using one of the current nightly versions (in contrast to the regular release version), you can try the following as a workaround: (Untested, as I do not have a Locus installation)

- Go to settings => system => view settings
- Activate "edit mode" (by tapping on the pencil button in the title bar)
- Search for "mapDirectory" (list is sorted alphabetically)
- Hit the "edit" button, enter the path you need, save and restart the map

**Comment by Swoddler** [comment]
Thank you. I have been on regular version. Have just signed up as beta tester if that's the same thing as getting the 'nightly' version? No updates have come with that sign up as yet. Excuse my lack of tech knowhow!

**Comment by moving-bits** [comment]
No problem.

Beta versions are something different. We have
- releases (typically every one or two months),
- most of them are preceded by one or two beta versions a few weeks before release
- and we do have the nightly builds prepared each night.

Nightly builds have to be installed manually (IIRC), and may contain errors and wrong development paths. They are intended for those who are tech-oriented and either want to have the cutting edge version of c:geo and/or want to contribute by testing and providing feedback.

**Comment by Swoddler** [comment]
I have just read the disclaimer on the nightly builds page....LOL. I perhaps need to take a different caching app for offline mapping on my travels when these 'air bridges' open next week...….

**Comment by Lineflyer** [comment]
@Swoddler 
Nightlies and beta versions are different things.
The nightlies are early alpha versions which come without any prior testing. Its in fact a snapshot of each nights development status.

Beta versions via Google Play are pretested versions which are published after some testing and bugfixing prior to that.
Currently there is no beta version active. You will receive the update to them as soon as a new beta is published.

**Comment by Lineflyer** [comment]
Answer from locus team:

> Hi xx,
sorry for this bug. We've redesigned the file browser but forgot to change its calling from external apps and addons. Will be fixed in the following version 3.47.

Release date not yet known.

**Comment by Lineflyer** [comment]
Addon: Release of the mentioned new Locus version is planned for next week.

**Comment by Lineflyer** [comment]
Solved by Locus version 3.47.1 released today.
